tests/marine/test_marine.py

```diff
@@ -2,85 +2,389 @@
 Note: in order to run the tests, you must put libmarine.so next to the marine_fixtures.py file
 """
 import pytest
+from typing import List, Union
+from marine import Marine
 
+from pypacker.layer12 import ethernet, arp
+from pypacker.layer3 import ip, icmp
+from pypacker.layer4 import tcp, udp
+from pypacker.layer567 import dns, http, dhcp
 
-# TODO: test more common protocols: ARP, ICMP, HTTP, etc.
 
+# TODO: Add a test for FTP.
 
-def test_tcp_filter_and_parse(marine_instance, tcp_packet, mac_1, mac_2, ip_1, ip_2, port_1, port_2,
-                              extracted_fields_from_tcp_packet):
-    expected = dict(zip(extracted_fields_from_tcp_packet, map(str, [mac_1, mac_2, ip_1, ip_2, port_1, port_2])))
 
-    passed, output = marine_instance.filter_and_parse(tcp_packet, 'ip', 'tcp', extracted_fields_from_tcp_packet)
+def general_filter_and_parse_test(
+    marine_instance: Marine,
+    packet: bytes,
+    bpf_filter: str,
+    display_filter: str,
```

Comment:
In lines 322 and 323 you're passing `bpf_filter` and `display_filter` as `None`, so this should be `Optional[str]`.